### ISSUE TITLE
Changes to the code from thea/queueregistry-demo

### ISSUE THREAD

**Issue by jcfreeman2** [DUNE-DAQ/app-framework-base#26]
I'm going to make the following changes to code merged into develop from thea/queueregistry-demo:

-Simplify NamedObject to just consist of a pure virtual function called "get_name" and to not contain a name_ member string; among other things this means that classes which inherit from it don't need to worry about kicking the string up through all the constructors of the classes in the inheritance chain.

-Get the code to follow the DUNE C++ Style Guidelines (https://github.com/DUNE-DAQ/styleguide/blob/develop/dune-daq-cppguide.md)

**Comment by mroda88** [comment]
Hi John, I was about to implement the second thing on my own while fixing some issues about the ERS. So I agree with both things. I'll leave the fix of the reference to you then and I will only fix the ERS issues.

**Comment by alessandrothea** [comment]
Hi John, Marco,
A general warning first. The code in develop is in-flux in preparation for v1. Another PR from Eric is expected on the timescale of tomorrow, which will be followed by a collapse of app-framework and app-framework-base in a single repo. Once that is done, and maybe another round of cleanup, the code will be in a better shape for code style review.

Now, about the specific comments:
- `NamedObject` is actually meant not to be abstract. The name in question is meant to be assigned at configuration time and remain un-muted for the lifecycle of the object. An abstract `get_name` would leave too much freedom to the object in question to return an arbitrary name

- `QueueRegistry::get()` fair enough. Shall we queue it among the fixes for v1? To this end I think it's time to start using github milestones and issue labels. I've created a milestone `v1.0`. Shall we add a `code style` issue label as well?

**Comment by mroda88** [comment]
I did the labelling bit. I agree, let's do the merge tomorrow and we can easily fix the rest after that
